### ISSUE TITLE
support upstream lua filter

### ISSUE THREAD

**Issue by bdecoste** [solo-io/gloo#3465]
**Is your feature request related to a problem? Please describe.**
Gloo/GlooE does not currently provide or support the upstream lua filter

**Describe the solution you'd like**
support upstream lua filter

**Comment by mjaow** [comment]
yes, I think this feature is very useful. We want to do some validation before request reaching user pod and response different based on user request.   With lua filter, I can customize the validation and response logic

**Comment by pvrdatadome** [comment]
Does it mean in the gateway configmap we can't use http_filters and envoy.filters.http.lua ?

**Comment by dspe** [comment]
Does it mean in the gateway configmap we can't use http_filters and envoy.filters.http.lua ?

**Comment by day0ops** [comment]
@DuncanDoyle theres been few asks for this lately.
